### ISSUE TITLE
Expand `OBJECT`s during view creation to include subfields in metadata

### ISSUE THREAD

**Issue by hammerhead** [crate/crate#16149]
### Problem Statement

A view's metadata currently does not expand `OBJECT`s to show fields from within the `OBJECT`:

```sql
cr> CREATE TABLE with_object (a OBJECT AS (b INTEGER));
CREATE OK, 1 row affected (0.216 sec)
cr> CREATE VIEW view_with_object AS SELECT a FROM with_object;
CREATE OK, 1 row affected (0.018 sec)
```

The table itself has the columns `a` and `a['b']`:
```sql
cr> SELECT column_name, data_type FROM information_schema.columns WHERE table_name = 'with_object';
+-------------+-----------+
| column_name | data_type |
+-------------+-----------+
| a           | object    |
| a['b']      | integer   |
+-------------+-----------+
SELECT 2 rows in set (0.004 sec)
```

The view, however, is missing information on `a['b']` and only shows the `OBJECT` itself:
```sql
cr> SELECT column_name, data_type FROM information_schema.columns WHERE table_name = 'view_with_object';
+-------------+-----------+
| column_name | data_type |
+-------------+-----------+
| a           | object    |
+-------------+-----------+
SELECT 1 row in set (0.007 sec)
```

### Possible Solutions

Retrieve the complete `OBJECT` structure during view creation and make it part of the view's metadata.

### Considered Alternatives

If it is a known limitation that cannot be lifted, add it to the documentation.

I could only find [this remark](https://cratedb.com/docs/crate/reference/en/latest/sql/statements/create-view.html#description):
> If a * is used to select the columns within the views query definition, this * will be resolved at query time like the rest of the query definition. This means if columns are added to the table after the view had been created, these columns will show up in subsequent queries on the view. It is generally recommended to avoid using * in view definitions.

Which does not seem to apply anymore and new columns show up immediately?
```sql
cr> CREATE TABLE t1 (a INTEGER);
CREATE OK, 1 row affected (0.212 sec)
cr> CREATE OR REPLACE VIEW v1 AS SELECT * FROM t1;
CREATE OK, 1 row affected (0.045 sec)
cr> ALTER TABLE t1 ADD COLUMN b INTEGER;
ALTER OK, -1 rows affected (0.052 sec)
cr> SELECT column_name, data_type FROM information_schema.columns WHERE table_name = 'v1';
+-------------+-----------+
| column_name | data_type |
+-------------+-----------+
| a           | integer   |
| b           | integer   |
+-------------+-----------+
SELECT 2 rows in set (0.008 sec)
```

**Comment by BaurzhanSakhariev** [comment]
Regarding remark - I think it's fine.

> Which does not seem to apply anymore and new columns show up immediately?

It doesn't say that new columns won't show up immediately. 
It's rather caveat that number of columns returned by `SELECT *` is not consistent  - and it can be dangerous for some user applications.

Regarding sub-columns not showing up: I will take a closer, look. 
We indeed have to update something: either docs or information_schema

**Comment by jeeminso** [comment]
A workaround:
```
cr> CREATE VIEW view_with_object AS SELECT a, a['b'] FROM with_object;                                                                                              
CREATE OK, 1 row affected  (0.089 sec)
cr> SELECT column_name, data_type FROM information_schema.columns WHERE table_name = 'view_with_object'; 
                                                                                                                                                                      
+-------------+-----------+
| column_name | data_type |
+-------------+-----------+
| a           | object    |
| a['b']      | integer   |
+-------------+-----------+
SELECT 2 rows in set (0.004 sec)
```

**Comment by BaurzhanSakhariev** [comment]
> A workaround:
> 

Thanks for pointing it out! I added it to the change entry.

[Fix](https://github.com/crate/crate/pull/16165) would cover situations, when a user accessing view doesn't know all columns (or sub-columns can be added dynamically and will show up in the `information_schema`).

**Comment by BaurzhanSakhariev** [comment]
Hi @hammerhead, thanks for reporting. 

Fix will be available in 5.7.3 hotfix release. 
See also doc changes in https://github.com/crate/crate/pull/16165
